target_postgres/tests/tests.py

```diff
@@ -244,4 +250,32 @@ def test_record_to_csv_row(record: dict, expected: list, dbsync_class: DbSync):
 )
 def test_record_primary_key_string(record, key_props: list, expected: str, dbsync_class: DbSync):
     dbsync_class.stream_schema_message['key_properties'] = key_props
-    assert dbsync_class.record_primary_key_string(record) == expected
\ No newline at end of file
+    assert dbsync_class.record_primary_key_string(record) == expected
+
+
+@pytest.mark.parametrize(
+    'types, expected',
+    [
+        # string always wins
+        (sorted(tuple(JSONSCHEMA_TYPES)), 'string'),
+        (('string', 'number'), 'string'),
+
+        (('integer', 'number'), 'number'),
+        (('boolean', 'integer', 'number'), 'number'),
+        (('boolean', 'integer'), 'integer'),
+
+        (('null', 'string'), 'string'),
+        (('array', 'object'), 'object'),
+
+        # None of these types generalize to each other, so we need to choose string as the general type
+        (('array', 'null', 'boolean'), 'string'),
+        # We don't know about these thypes, so again assume string is best
+        (('fake type 1', 'fake type 2', 'fake type 3'), 'string'),
```

Comment:
Can we also include test cases with a combination of fake types and supported types?